src/MDOP.jl

```diff
@@ -4,16 +4,150 @@ using Distances
+    allNNidxs, NNdist_old = all_neighbors(vtree, Y_act, 1:length(Y_act), 1, w)
+
+    # set a flag, in order to tell the while loop when to stop. Each loop
+    # stands for encountering a new embedding dimension
+    flag = true
+
+    # set index-counter for the while loop
+    cnt = 1;
+
+    # preallocate output variables
+    τ_vals = zeros(Int,1)
+    ts_vals = ones(Int,1)
+    FNNs = zeros(Float64,1)
+    if β_condition; βS = Array{T}(undef, length(τs), max_num_of_cycles); end
+
+    # loop over increasing embedding dimensions until some break criterion will
+    # tell the loop to stop/break
+    while flag
+
+        # get the β-statistic
+        β = beta_statistic(Y_act, s; τs = τs , w = w)
+        if β_condition; βS[:, cnt] = β; end
+
+        # determine the optimal tau value from the β-statistic
+        maxi, max_idx = findmax(β)
+        # store chosen delay (and chosen time series)
+        push!(τ_vals, τs[max_idx])
+        push!(ts_vals, 1)
+
+        # create phase space vector for this embedding cycle
+        Y_act = DelayEmbeddings.embed_one_cycle(Y_act,s,τ_vals[cnt+1])
+
+        # compute nearest neighbor distances in the new embedding dimension for
+        # FNN statistic
+        vtree = KDTree(Y_act, metric)
+        allNNidxs, NNdist_new = all_neighbors(vtree, Y_act, 1:length(Y_act), 1, w)
+
+        # truncate distance-vector to the "new" length of the actual embedding vector
+        NNdist_old = NNdist_old[1:length(Y_act)]
+
+        # compute FNN-statistic and store vals
+        fnns = fnn_embedding_cycle(NNdist_old,NNdist_new;r=r)
+        cnt == 1 ? FNNs[1] = fnns : push!(FNNs,fnns)
+
+        # break criterion 1
+        if FNNs[cnt] <= fnn_thres
+            flag = false
+            display("Algorithm stopped due to sufficiently small FNNs. VALID embedding achieved.")
+        end
+        # break criterion 2
+        if cnt > 1 && FNNs[cnt]>FNNs[cnt-1]
+            flag = false
+            display("Algorithm stopped due to rising FNNs.")
+        end
+        # break criterion 3 (maximum embedding cycles reached)
+        if max_num_of_cycles == cnt; flag = false; end
+
+        cnt += 1
+        # for the next embedding cycle save the distances of NN for this dimension
+        NNdist_old = NNdist_new
+    end
+
+    if β_condition
+        return Y_act[:,1:cnt-1], τ_vals[1:cnt-1], ts_vals[1:cnt-1], FNNs[1:cnt-1], βS[:,1:cnt-1]
+    else
+        return Y_act[:,1:cnt-1], τ_vals[1:cnt-1], ts_vals[1:cnt-1], FNNs[1:cnt-1]
+    end
+end
+
+"""
+    MDOP(s::Dataset; kwargs...) → Y, τ_vals, ts_vals, FNNs
+MDOP is a unified approach to properly embed a time series based on the paper of
+Chetan Nichkawde [^Nichkawde2013].
+[`uzal_cost`](@ref)
+
+
+[^Nichkawde2013]: Nichkawde, Chetan (2013). [Optimal state-space reconstruction using derivatives on projected manifold. Physical Review E 87, 022905](https://doi.org/10.1103/PhysRevE.87.022905).
+"""
+function MDOP()
     #tbd
 
 end
```

Comment:
Do we need this code? Seems to be left-over that can be deleted.

Reply:
This is a place marker for the multivariate embedding method. So far, we just considered univariate input. This is going to be a new PR then (same holds for #38 ).